forte/processors/base/data_augment_processor.py

```diff
@@ -46,44 +55,559 @@ class ReplacementDataAugmentProcessor(BaseDataAugmentProcessor):
-        It consists of tuples (entry, new text) inserted by :func: replace.
-        The new text will be used for building new data pack.
+        r"""
+        - replaced_annos: {datapack id: List[span, text]}
+            It records the spans replaced by new texts.
+            It is a map from datapack id to a list of tuples
+            (span, new text) inserted by :func: replace.
+            The new text will be used for building new data pack.
+
+        - inserted_annos: {datapack id: Dict{position: length}}
+            It records the inserted spans, mapping from datapack id
+            to a dictionary (position -> length) inserted
+            by :func: insert.
+            The position is the index in the original datapack
+            of insertion, and the length is the length of the
+            inserted string.
+
+        - deleted_annos: {datapack id: Set[annotation tid]}
+            It records the deleted spans, mapping from datapack id
+            to a set of annotation tids appended by :func: delete.
+
+        - data_pack_map: {orig pack id: new pack id}
+            It maintains a mapping from the pack id
+            of the original pack to the pack id of augmented pack.
+            It is used when copying the MultiPackLink.
+
+        - anno_maps: {datapack id: Dict{orig tid, new tid}}
+            It is a map for tracking the annotation ids
+            before and after the auto align. It maps the
+            original annotation tid to the new annotation tid.
+            It is used when copying the MultiPackLink.
         """
         super().__init__()
-        self.replaced_spans: List[Tuple[Entry, str]] = []
+        self.replaced_annos: DefaultDict[int, List[Tuple[Span, str]]] = \
+            defaultdict(list)
+        self.inserted_annos: DefaultDict[int, Dict[int, int]] = \
+            defaultdict(dict)
+        self.deleted_annos: DefaultDict[int, Set[int]] = defaultdict(set)
+
+        self.data_pack_map: Dict[int, int] = {}
+        self.anno_maps: Dict[int, Dict[int, int]] = {}
+        self.other_entry_policy: Dict[str, str] = {}
+
+    def initialize(self, resources: Resources, configs: Config):
+        super().initialize(resources, configs)
+        # Parse the "other_entry_policy" in the configuration.
+        self.other_entry_policy = self._get_other_entry_policy_dict()
 
-    def replace(self, replacement_op: TextReplacementOp, input: Entry):
+    def _get_other_entry_policy_dict(self) -> Dict[str, str]:
+        r"""
+        A helper function for parsing the "other_entry_policy"
+        configuration.
         """
+        entries = self.configs['other_entry_policy']["entry"]
+        policies = self.configs['other_entry_policy']["policy"]
+        other_entry_policy = {}
+        for entry, policy in zip(entries, policies):
+            other_entry_policy[entry] = policy
+        return other_entry_policy
+
+    def _is_span_overlap(self, pid: int, begin: int, end: int) -> bool:
```

Comment:
The function name is different from its purpose. Something like `__overlap_with_existing` would be more accurate.